src/main/java/hudson/tasks/junit/History.java

```diff
@@ -512,22 +442,20 @@ private ObjectNode computeDistributionJson(List<HistoryTestResultSummary> histor
                 minDuration = h.getDuration();
             }
         }
-        double extraDuration = Math.max(0.001, (maxDuration - minDuration) * 0.05);
-        minDuration = Math.max(0.0, minDuration - extraDuration);
-        maxDuration = maxDuration + extraDuration;
-        int[] counts = new int[100];
-        int smoothBuffer = 2;
-        double[] lrX = new double[counts.length + smoothBuffer * 2 + 1];
-        double[] lrY = new double[counts.length + smoothBuffer * 2 + 1];
-        double scale = maxDuration - minDuration;
-        double step = scale / counts.length;
+        // double extraDuration = Math.max(0.001, (maxDuration - minDuration) * 0.05);
+        minDuration = Math.max(0.0, minDuration);
+        int buckets = 50;
+        double[] lrX = new double[buckets];
+        double[] lrY = new double[buckets];
+        double domain = maxDuration - minDuration;
+        double step = domain / buckets;
         for (HistoryTestResultSummary h : history) {
-            int idx = smoothBuffer + (int) Math.round((h.getDuration() - minDuration) / step);
+            int idx = (int) Math.round((h.getDuration() - minDuration) / step);
             int idx2 = Math.max(0, Math.min(idx, lrY.length - 1));
             lrY[idx2]++;
         }
         for (int i = 0; i < lrY.length; ++i) {
-            lrX[i] = ((minDuration + (maxDuration - minDuration) / lrY.length * i) / scale * 100.0);
+            lrX[i] = minDuration + step * (i + 0.5);
```

Comment:
Offset the X values so they are in the middle of the bucket. We are no longer scaling them into the range 0-1 because we are using them directly in our data series.